dashboard/test/controllers/pd/workshop_certificate_controller_test.rb

```diff
@@ -44,10 +44,14 @@ class Pd::WorkshopCertificateControllerTest < ::ActionController::TestCase
     workshop = create :csf_intro_workshop
     enrollment = create :pd_enrollment, :with_attendance, workshop: workshop
 
-    _ = anything
-    mock_draw = expect_renders_certificate
-    mock_draw.expects(:annotate).with(_, _, _, _, _, 'CS Fundamentals')
-    mock_draw.expects(:annotate).with(_, _, _, _, _, 'Intro Workshop')
+    mock_image = expect_renders_certificate
+    expect_renders_text(mock_image, enrollment.full_name)
+    expect_renders_text(mock_image, 'CS Fundamentals')
+    expect_renders_text(mock_image, 'Intro Workshop')
+    workshop.facilitators.each {|f| expect_renders_text(mock_image, f.name)}
+    workshop_hours = Integer(workshop.effective_num_hours)
+    expect_renders_text(mock_image, workshop_hours.to_s)
```

Comment:
is there a reason this isn't just

```
expect_renders_text(mock_image, workshop.effective_num_hours.to_s)
```

Reply:
Because then it would render as `6.0` but the certificate is supposed to display an integer `6`.
The rendering logic does some rounding too. I tried copy/pasting exactly what the implementation code did but it always resulted in `nil` rather than `6`. I'm not sure why it didn't work when I used in in my test case:
https://github.com/code-dot-org/code-dot-org/blob/378ae9629385a9e2153571c1d5b3682a8f541851/dashboard/lib/pd/certificate_renderer.rb#L69

Reply:
oh, that's super funky. Might be worth adding a comment to this test explaining what's going on here and linking to that code